### PR TITLE
introduce CircleCI config.yml generation using typescript sdk

### PR DESCRIPTION
**Pull request by Okhelifi**
## Issue

https://gravitee.atlassian.net/browse/APIM-2635

## Description

In this pull request we introduce the circle config.yml generation using the typescript sdk. To se the result please refer to the `pipeline-package-bundle.spec.ts` and the associated resources.

### CODE REVIEW

**Review by phiz71 — APPROVED** [review]

File: .circleci/ci/src/config.ts [hunk 1]
```diff
@@ -0,0 +1,88 @@
+import { DockerResourceClass } from '@circleci/circleci-config-sdk/dist/src/lib/Components/Executors/types/DockerExecutor.types';
+
+// const cache = {
+//   prefix: 'gravitee-api-management-v9',
+// };
```
Comment (ytvnr) [inline .circleci/ci/src/config.ts:5]: to remove ?

File: .circleci/ci/README.md [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+### Generate CircleCI config locally:
+
+```bash
+#export CIRCLE_BRANCH=my-awesome-branch-🦄
+#export GIT_BASE_BRANCH=master
+export CI_ACTION=pull_requests
+#export CIRCLE_TAG=4.1.0
+export CI_TAG=4.1.0
+export CIRCLE_SHA1=784ff35ca
```
Comment (ytvnr) [inline .circleci/ci/README.md:42]: I would not comment line if it's for documentation :)

File: .circleci/ci/src/config.ts [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+  openjdk: {
+    image: 'cimg/openjdk',
+    resource: 'medium' as DockerResourceClass,
+    version: '17.0',
+  },
+};
+
+// const maven = {
+//   settingsFile: '/tmp/.gravitee.settings.xml',
+// };
```
Comment (ytvnr) [inline .circleci/ci/src/config.ts:30]: useless ?

File: .circleci/ci/src/config.ts [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+  github: '1.0.5',
+  helm: '2.0.1',
+  keeper: '0.6.3',
+  slack: '4.12.5',
+  snyk: '1.7.0',
+};
+
+// const remoteDocker = {
+//   version: '20.10.6',
+// };
```
Comment (ytvnr) [inline .circleci/ci/src/config.ts:45]: to remove ?

File: .circleci/ci/tsconfig.json [hunk 5]
```diff
@@ -0,0 +1,30 @@
+{
+    "compilerOptions": {
+        "incremental": true,
+
+        /* Language and Environment */
+        "target": "es2022",
+        // "lib": [],
```
Comment (ytvnr) [inline .circleci/ci/tsconfig.json:7]: to remove ?

File: .circleci/ci/src/config.ts (hunk 1 above)
Comment (phiz71) [reply .circleci/ci/src/config.ts:5]: Will be uncomment later when used

File: .circleci/ci/src/config.ts (hunk 3 above)
Comment (phiz71) [reply .circleci/ci/src/config.ts:30]: Will be uncomment later when used

File: .circleci/ci/src/config.ts (hunk 4 above)
Comment (phiz71) [reply .circleci/ci/src/config.ts:45]: Will be uncomment later when used

File: .circleci/ci/README.md (hunk 2 above)
Comment (Okhelifi) [reply .circleci/ci/README.md:42]: ok

**Pull request merged**